.github/workflows/rust_ci.yml

```diff
@@ -28,7 +28,7 @@ jobs:
         run: cargo test --all-features
 
       - name: Cargo build
-        run: cargo build
+        run: cargo build --all-features
```

Comment:
So we also want to do a check that we can't have a feature combination that causes build/check to fail (which can sometimes happen as you add more features) Given the simplicity of the project so far we could probably just get away with  a `cargo build --no-default-features` as serde missing won't break anything